### PR TITLE
feat(search-response-docs): adds docs

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request adds docs to the search response type.

### CODE REVIEW

File: packages/client-search/src/types/SearchResponse.ts [hunk 1]
```diff
@@ -1,42 +1,175 @@
...
+
+  /**
+   * Index of the current page (zero-based).
+   */
   readonly page: number;
+
+  /**
+   * Number of hits returned (used only with offset)
+   */
   readonly length?: number;
```
Comment (Haroenv) [inline packages/client-search/src/types/SearchResponse.ts]: we could have a discriminated union here: `baseParams & (hitsPerPageParams | offsetParams)`

File: packages/client-search/src/types/SearchResponse.ts [hunk 2]
```diff
@@ -1,42 +1,175 @@
...
+
+  /**
+   * Used to return warnings about the query.
+   */
   readonly message?: string;
+
+  /**
+   * The computed geo location.
+   *
+   * Format: ${lat},${lng}, where the latitude and longitude are expressed as decimal floating point number.
```
Comment (Haroenv) [inline packages/client-search/src/types/SearchResponse.ts]:
```suggestion
   * Format: "lat,lng", where the latitude and longitude are expressed as decimal floating point number.
```

this is also clear enough imo

File: packages/client-search/src/types/SearchResponse.ts (hunk 1 above)
Comment (nunomaduro) [reply packages/client-search/src/types/SearchResponse.ts]: I don't understand your comment.

Comment (Haroenv) [reply packages/client-search/src/types/SearchResponse.ts]: ` { hits, ...} & ({ page, hitsPerPage} | { offset, length })`

something like that

**Comment by Haroenv** [comment]
Not necessary, but they show up in the editor when you're typing, so that's good for DX

File: packages/client-search/src/types/SearchResponse.ts (hunk 1 above)
Comment (nunomaduro) [reply packages/client-search/src/types/SearchResponse.ts]: I won't have any type like that yet. Let's leave it for maybe next time.

**Pull request merged**